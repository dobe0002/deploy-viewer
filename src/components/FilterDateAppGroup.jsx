/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import PropTypes from 'prop-types';
import DatePicker from 'react-datepicker';
import styled from 'styled-components';
import { FormGroup, Label, Form, Card, CardBody, Row, Col } from 'reactstrap';
import SelectAppGroup from './SelectAppGroup';

export default class FilterDateAppGroup extends React.Component {
  constructor(props) {
    super(props);

    const { startDate, endDate, currentAppGroup } = this.props;

    this.state = {
      startDate,
      endDate,
      currentAppGroup
    };
    this.handleStartChange = this.handleStartChange.bind(this);
    this.handleEndChange = this.handleEndChange.bind(this);
    this.handleAppGroupChange = this.handleAppGroupChange.bind(this);
  }

  handleStartChange(date) {
    const { onStartDateChange } = this.props;
    this.setState({ startDate: date });
    onStartDateChange(date);
  }

  handleEndChange(date) {
    const { onEndDateChange } = this.props;

    this.setState({ endDate: date });
    onEndDateChange(date);
  }

  handleAppGroupChange(newAppGroup) {
    const { onAppGroupChange } = this.props;
    this.setState({ currentAppGroup: newAppGroup });
    onAppGroupChange(newAppGroup);
  }

  render() {
    const { startDate, endDate, currentAppGroup } = this.state;
    const { appGroups, dataTitle } = this.props;
    const NavHeader = styled.div`
      input {
        height: calc(1.5em + 0.75rem + 2px);
        border: 1px solid #ced4da;
        border-radius: 0.25rem;
      }
      label {
        font-weight: bold;
        padding-right: 10px;
        &:after {
          content: ':';
        }
      }
      .form-group {
        margin-right: 30px;
      }
      margin-top: 20px;
      margin-bottom: 20px;
    `;
    return (
      <NavHeader>
        <Row>
          {dataTitle !== '' && (
            <Col xs='3'>
              <h1>{dataTitle}</h1>
            </Col>
          )}
          <Col>
            <Card>
              <CardBody>
                <Form inline>
                  <FormGroup>
                    <Label for='timelineStartDate'>Start</Label>

                    <DatePicker
                      selected={startDate}
                      onChange={this.handleStartChange}
                      maxDate={endDate}
                      id='timelineStartDate'
                    />
                  </FormGroup>
                  <FormGroup>
                    <Label for='timelineEndDate'>End</Label>
                    <DatePicker
                      selected={endDate}
                      onChange={this.handleEndChange}
                      minDate={startDate}
                      id='timelineEndDate'
                    />
                  </FormGroup>

                  <SelectAppGroup
                    appGroups={appGroups}
                    getChange={this.handleAppGroupChange}
                    currentAppGroup={currentAppGroup}
                  />
                </Form>
              </CardBody>
            </Card>
          </Col>
        </Row>
      </NavHeader>
    );
  }
}

FilterDateAppGroup.propTypes = {
  dataTitle: PropTypes.string,
  startDate: PropTypes.instanceOf(Date),
  endDate: PropTypes.instanceOf(Date),
  currentAppGroup: PropTypes.string,
  appGroups: PropTypes.objectOf,
  onStartDateChange: PropTypes.func,
  onEndDateChange: PropTypes.func,
  onAppGroupChange: PropTypes.func
};

FilterDateAppGroup.defaultProps = {
  dataTitle: '',
  startDate: new Date(),
  endDate: new Date(),
  currentAppGroup: '',
  appGroups: {},
  onStartDateChange: () => {},
  onEndDateChange: () => {},
  onAppGroupChange: () => {}
};
