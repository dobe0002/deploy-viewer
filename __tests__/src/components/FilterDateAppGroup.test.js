/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilterDateAppGroup from '../../../src/components/FilterDateAppGroup';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('FilterDateAppGroup  tests', () => {
  let startDate = '';
  let endDate = '';
  let currentAppGroup = '';
  const setStartDateMock = date => {
    startDate = date;
  };
  const setEndDateMock = date => {
    endDate = date;
  };
  const setAppGroupMock = appGroup => {
    currentAppGroup = appGroup;
  };
  const sampleProps = {
    startDate: new Date('2019-02-01T06:00:00.000Z'),
    endDate: new Date('2019-03-02T06:00:00.000Z'),
    currentAppGroup: '',
    appGroups: {},
    onStartDateChange: setStartDateMock,
    onEndDateChange: setEndDateMock,
    onAppGroupChange: setAppGroupMock
  };

  beforeEach(() => {
    startDate = '';
    endDate = '';
    currentAppGroup = '';
  });

  test('Accessibility check', async () => {
    const wrapper = shallow(
      <main>
        <FilterDateAppGroup />
      </main>
    ); // Main tag is needed to prevent landmark errors
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });
  test('Snapshot test', () => {
    const wrapper = shallow(
      <FilterDateAppGroup
        startDate={sampleProps.startDate}
        endDate={sampleProps.endDate}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('Call handleStartChange', () => {
    expect(startDate).toEqual('');
    const wrapper = shallow(
      <FilterDateAppGroup
        onStartDateChange={sampleProps.onStartDateChange}
        startDate={sampleProps.startDate}
      />
    );
    expect(wrapper.state('startDate')).toEqual(sampleProps.startDate);

    const newStartTestDate = new Date(2019, 11, 12);
    wrapper.instance().handleStartChange(newStartTestDate);
    expect(wrapper.state('startDate')).toEqual(newStartTestDate);
    expect(startDate).toEqual(newStartTestDate);
  });
  test('Call handleEndChange', () => {
    expect(endDate).toEqual('');
    const wrapper = shallow(
      <FilterDateAppGroup
        onEndDateChange={sampleProps.onEndDateChange}
        endDate={sampleProps.endDate}
      />
    );
    expect(wrapper.state('endDate')).toEqual(sampleProps.endDate);

    const newEndTestDate = new Date(2019, 11, 22);
    wrapper.instance().handleEndChange(newEndTestDate);
    expect(wrapper.state('endDate')).toEqual(newEndTestDate);
    expect(endDate).toEqual(newEndTestDate);
  });
  test('Call handleAppGroupChange', () => {
    expect(currentAppGroup).toEqual('');
    const wrapper = shallow(
      <FilterDateAppGroup
        onAppGroupChange={sampleProps.onAppGroupChange}
        currentAppGroup='Foo'
      />
    );
    expect(wrapper.state('currentAppGroup')).toEqual('Foo');

    wrapper.instance().handleAppGroupChange('Boo');
    expect(wrapper.state('currentAppGroup')).toEqual('Boo');
    expect(currentAppGroup).toEqual('Boo');
  });
});
