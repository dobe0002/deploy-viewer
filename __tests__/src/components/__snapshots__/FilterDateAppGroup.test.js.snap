// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDateAppGroup  tests Snapshot test 1`] = `
<styled.div>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Card
        tag="div"
      >
        <CardBody
          tag="div"
        >
          <Form
            inline={true}
            tag="form"
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="timelineStartDate"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Start
              </Label>
              <t
                allowSameDay={false}
                dateFormat="MM/dd/yyyy"
                dateFormatCalendar="LLLL yyyy"
                disabled={false}
                disabledKeyboardNavigation={false}
                dropdownMode="scroll"
                id="timelineStartDate"
                inlineFocusSelectedMonth={false}
                maxDate={2019-03-02T06:00:00.000Z}
                monthsShown={1}
                nextMonthButtonLabel="Next Month"
                nextYearButtonLabel="Next Year"
                onBlur={[Function]}
                onCalendarClose={[Function]}
                onCalendarOpen={[Function]}
                onChange={[Function]}
                onClickOutside={[Function]}
                onFocus={[Function]}
                onInputClick={[Function]}
                onInputError={[Function]}
                onKeyDown={[Function]}
                onMonthChange={[Function]}
                onSelect={[Function]}
                onYearChange={[Function]}
                preventOpenOnFocus={false}
                previousMonthButtonLabel="Previous Month"
                previousYearButtonLabel="Previous Year"
                readOnly={false}
                renderDayContents={[Function]}
                selected={2019-02-01T06:00:00.000Z}
                shouldCloseOnSelect={true}
                showMonthYearPicker={false}
                showPopperArrow={true}
                showPreviousMonths={false}
                showQuarterYearPicker={false}
                showTimeInput={false}
                showTimeSelect={false}
                strictParsing={false}
                timeCaption="Time"
                timeInputLabel="Time"
                timeIntervals={30}
                withPortal={false}
              />
            </FormGroup>
            <FormGroup
              tag="div"
            >
              <Label
                for="timelineEndDate"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                End
              </Label>
              <t
                allowSameDay={false}
                dateFormat="MM/dd/yyyy"
                dateFormatCalendar="LLLL yyyy"
                disabled={false}
                disabledKeyboardNavigation={false}
                dropdownMode="scroll"
                id="timelineEndDate"
                inlineFocusSelectedMonth={false}
                minDate={2019-02-01T06:00:00.000Z}
                monthsShown={1}
                nextMonthButtonLabel="Next Month"
                nextYearButtonLabel="Next Year"
                onBlur={[Function]}
                onCalendarClose={[Function]}
                onCalendarOpen={[Function]}
                onChange={[Function]}
                onClickOutside={[Function]}
                onFocus={[Function]}
                onInputClick={[Function]}
                onInputError={[Function]}
                onKeyDown={[Function]}
                onMonthChange={[Function]}
                onSelect={[Function]}
                onYearChange={[Function]}
                preventOpenOnFocus={false}
                previousMonthButtonLabel="Previous Month"
                previousYearButtonLabel="Previous Year"
                readOnly={false}
                renderDayContents={[Function]}
                selected={2019-03-02T06:00:00.000Z}
                shouldCloseOnSelect={true}
                showMonthYearPicker={false}
                showPopperArrow={true}
                showPreviousMonths={false}
                showQuarterYearPicker={false}
                showTimeInput={false}
                showTimeSelect={false}
                strictParsing={false}
                timeCaption="Time"
                timeInputLabel="Time"
                timeIntervals={30}
                withPortal={false}
              />
            </FormGroup>
            <SelectAppGroup
              appGroups={Object {}}
              currentAppGroup=""
              getChange={[Function]}
            />
          </Form>
        </CardBody>
      </Card>
    </Col>
  </Row>
</styled.div>
`;
